src/services/localSSHService.ts

```diff
@@ -44,12 +47,31 @@ export class LocalSSHService extends Disposable implements ILocalSSHService {
                     // event, so ignore it if more settings are affected at the same time.
                     return;
                 }
-                if (this.initialized) {
-                    await this.initialized;
-                }
-                this.initialized = this.initialize();
+                await this.ensureInitialize();
             }
         }));
+        this.experimentService.onDidChangeFlag(async (key) => {
+            if (key === FeatureFlags.gitpod_desktop_use_local_ssh_proxy) {
+                await this.ensureInitialize();
+            }
+        });
+    }
+
+    public async tryInitialize() {
+        if (this.initialized) {
+            return this.initialized;
+        }
+        this.initialized = this.initialize();
+    }
+
+    private async ensureInitialize() {
+        if (!(await this.experimentService.getUseLocalSSHProxy())) {
+            return;
+        }
+        if (this.initialized) {
```

Comment:
I'm not sure it is right way doing it. 2 parallel calls will blcok on previous one and will run in parallel

probably you want a queue, something like that:
```
this.initialized = (async () => {
   await this.initialized;
   this.initialize();
});
```